src/converters/cover.ts

```diff
@@ -139,6 +165,12 @@ class CoverHandler implements ServiceHandler {
     this.service.updateCharacteristic(hap.Characteristic.CurrentPosition, characteristicValue);
   }
 
+  private scaleAndUpdateCurrentTilt(value: number): void {
+    // map value: percentages to characteristicValue: angle
+    const characteristicValue = -90 + (value / 100) * 180;
+    this.service.updateCharacteristic(hap.Characteristic.CurrentHorizontalTiltAngle, characteristicValue);
+  }
```

Comment:
If you only wish to update the characteristic based on the received MQTT value, I would suggest to use one of the monitor classes (from `src/converters/monitor.ts`). In this case, you'll probably need the `NumericCharacteristicMonitor`, which will also handle the scaling.

For a reference, have a look at how the Brightness characteristic in `light.ts` is handled (see `tryCreateBrightness` and `updateState` in that file).

Reply:
I'm trying to see the advantage here, what would it be?

Reply:
The main advantage would be to not duplicate logic/code.